src/app/page.tsx

```diff
@@ -1,24 +1,61 @@
-import { CategoriesExpandableList } from "@/components/CategoriesExpandableList";
-import { CategoriesMenu } from "@/components/CategoriesMenu";
-import Header from "@/components/Header";
+"use client";
+
+import { useState } from "react";
+import dynamic from "next/dynamic";
+
+import CategoriesExpandableList from "@/components/CategoriesExpandableList";
 import ListItemCard from "@/components/ListItemCard";
+import { Modal } from "flowbite-react";
+import PreviewItemsAddedSkeleton from "@/components/PreviewItemsAdded/PreviewItemAddedSkeleton";
+
+const PreviewItemAdded = dynamic(
+  () =>
+    import("@/components/PreviewItemsAdded/PreviewItemAdded").then(
+      (module) => module.default
+    ),
+  {
+    loading: () => <PreviewItemsAddedSkeleton />,
+  }
+);
+
+const items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 
 export default function Home() {
-  const items = [1,2,3,4,5,6,7,8,9,10]
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const handleAddToCart = () => {
```

Comment:
Keeping handler functions outside of the template simplifies the template, and enables the dev to write handler code without it being surrounded by other non related items